build/plugin-add-imports.ts

```diff
@@ -0,0 +1,26 @@
+// adapted from https://gist.github.com/HenrikJoreteg/38b2717c138b5997ae406b1e2d9c556d
+
+import type { FilterPattern } from '@rollup/pluginutils';
+import type { Plugin, TransformResult } from 'rollup';
+import { createFilter } from '@rollup/pluginutils';
+
+interface Options {
+  include: FilterPattern;
+  imports: string[];
+}
+
+export const addImports = (opts: Options): Plugin => {
+    const filter = createFilter(opts.include);
+
+    return {
+        name: 'add-imports',
+
+        transform(code: string, id: string): TransformResult {
+            if (!filter(id)) return;
```

Comment:
This plugin is not working as expected for me, `filter(id)` always returns `false` in my Windows 10 setup, even for the `index.ts` file which should be an exact match:
```js
opts = {
  include: 'D:\\Code\\MusicBrainz\\userscripts-ROpdebee\\src\\mb_enhanced_cover_art_uploads\\index.ts',
  imports: [ "import '@vendor/gm4-polyfill/gm4-polyfill'" ]
};
id = 'D:\\Code\\MusicBrainz\\userscripts-ROpdebee\\src\\mb_enhanced_cover_art_uploads\\index.ts';
```
Using a hard-coded string equality condition instead of the filter worked and gave me a built userscript which includes the polyfill.

The problem seems to be that `filter()` always does `normalizePath(id)` (which replaces `\` by `/`), while `createFilter()` does not normalize absolute paths for the include and exclude filter patterns. Below is a work-around for our use-case, the alternative would be to pass a relative path to the plugin:
```suggestion
export const addImports = (opts: Options): Plugin => {
    if (typeof opts.include === 'string') {
        opts.include = normalizePath(opts.include);
    }
    const filter = createFilter(opts.include);

    return {
        name: 'add-imports',

        transform(code: string, id: string): TransformResult {
            if (!filter(id)) return;
```
I don't know whether I have overseen some other possible use-cases, but I think this should probably be fixed in the '@rollup/pluginutils' package.

Reply:
Hmm, interesting. I actually just removed this code, so no need to fix it 😄